portability/toku_instrumentation.h

```diff
@@ -0,0 +1,47 @@
+/* -*- mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*- */
+// vim: ft=cpp:expandtab:ts=8:sw=4:softtabstop=4:
+#ident "$Id$"
+/*======
+  This file is part of PerconaFT.
+
+
+  Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved.
+
+  PerconaFT is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License, version 2,
+  as published by the Free Software Foundation.
+
+  PerconaFT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with PerconaFT.  If not, see <http://www.gnu.org/licenses/>.
+
+  ----------------------------------------
+
+  PerconaFT is free software: you can redistribute it and/or modify
+  it under the terms of the GNU Affero General Public License, version 3,
+  as published by the Free Software Foundation.
+
+  PerconaFT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with PerconaFT.  If not, see <http://www.gnu.org/licenses/>.
+  ======= */
+
+#ident "Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved."
+
+#pragma once
+
+#include <stdio.h> // FILE
+
+struct TOKU_FILE
```

Comment:
Proper header file organisation is still something we are trying to figure out. The current prototype has the following scheme: toku_portability.h includes toku_instrumentation.h (declaring instrumentation stub structs and functions), which optionally includes toku_mysql.h (declaring same structs and functions for MySQL PFS).

TOKU_FILE falls into toku_instrumentation.h by the path of least resistance then. But the above scheme has things declared very much out of place (mutex stuff etc) to resolve cyclic dependencies.

The above means that the current header scheme is not final. I can try introducing toku_file.h and move all the file-related declarations there too at the same time.